### PR TITLE
Hotfix: Fixing app types variants (fixes Hubspot)

### PR DESCRIPTION
**Pull request by leog**
## What does this PR do?

HubSpot App from the appstore was not showing the install button. This fixes that and other eventual apps in the same situation.

**Environment**: Staging(main branch) / Production

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

Go to https://localhost:3000/apps/hubspot and check the install button is present.

### CODE REVIEW

File: packages/lib/deriveAppDictKeyFromType.ts [hunk 1]
```diff
@@ -21,6 +21,12 @@ export function deriveAppDictKeyFromType(appType: string, dict: Record<string, u
     return appTypeVariant2;
   }
 
+  const appTypeVariant3 = appType.replace(/_/g, "");
+  handlers = dict[appTypeVariant3];
+  if (handlers) {
+    return appTypeVariant3;
+  }
+
```
Comment (PeerRich) [inline packages/lib/deriveAppDictKeyFromType.ts:29]: can you explain what this does? maybe with a comment above

**Review by hariombalhara — APPROVED** [review]
Ohh.. a special case for Hubspot. Sorry about that.

**Comment by leog** [comment]
> Ohh.. a special case for Hubspot. Sorry about that.

No worries :)

**Comment by hariombalhara** [comment]
I will add a test for it https://github.com/calcom/cal.com/pull/3176. Tracking it here.

File: packages/lib/deriveAppDictKeyFromType.ts (hunk 1 above)
Comment (zomars) [reply packages/lib/deriveAppDictKeyFromType.ts:29]: A comment on the code itself would be helpful @leog

Comment (leog) [reply packages/lib/deriveAppDictKeyFromType.ts:30]: Done as well 👍

**Pull request merged**